doc/interesting-queries.md

```diff
@@ -520,6 +520,23 @@ tx_id |units_mem|units_steps|redeemer_fee|purpose|                      policy
 (2 rows)
 
 ```
+
+### Get blocks with 0 transactions and the pools that forged it
+```sql
+select distinct on(block.hash) block.hash as block_hash , epoch_no, tx_count, pool_hash.hash_raw as pool_hash, pool_update.pledge,
+pool_update.active_epoch_no, pool_metadata_ref.url, pool_offline_data.ticker_name  from block join slot_leader on block.slot_leader_id = slot_leader.id
+join pool_hash on slot_leader.pool_hash_id = pool_hash.id join pool_update on pool_update.hash_id = pool_hash.id left join pool_metadata_ref on
+pool_update.meta_id = pool_metadata_ref.id left join pool_offline_data on pool_offline_data.pmr_id = pool_metadata_ref.id  where tx_count = 0 and
+epoch_no > 150  ORDER BY block.hash, pool_update.active_epoch_no desc;
```

Comment:
Maybe format this slightly?
```suggestion
select distinct on(block.hash) block.hash as block_hash , epoch_no, tx_count, pool_hash.hash_raw as pool_hash,
                               pool_update.pledge, pool_update.active_epoch_no, pool_metadata_ref.url, pool_offline_data.ticker_name  
  from block join slot_leader on block.slot_leader_id = slot_leader.id
    join pool_hash on slot_leader.pool_hash_id = pool_hash.id 
    join pool_update on pool_update.hash_id = pool_hash.id 
    left join pool_metadata_ref on pool_update.meta_id = pool_metadata_ref.id 
    left join pool_offline_data on pool_offline_data.pmr_id = pool_metadata_ref.id  
  where tx_count = 0 and epoch_no > 150  
  order by block.hash, pool_update.active_epoch_no desc;
```

Reply:
I wonder why the `distinct` needs to be there. To me, that suggests the query can be improved so that the `distinct` is not needed.

Reply:
A pool may have multiple pool updates. So without the disticts, each block may appear multiple times

Reply:
This is what the ` order by block.hash, pool_update.active_epoch_no desc;` is for. It brings the update with the latest `active_epoch_no`. Different orderings can be applied, this is a sample.